pkg/process/net/network_linux.go

```diff
@@ -33,11 +33,15 @@ func init() {
 
 // RemoteSysProbeUtil wraps interactions with a remote system probe service
 type RemoteSysProbeUtil struct {
+	mu *sync.Mutex
+
 	// Retrier used to setup system probe
 	initRetry retry.Retrier
 
 	socketPath string
 	httpClient http.Client
+
+	buf bytes.Buffer
```

Comment:
It would be good to group these fields together and/or add a comment noting that the buffer is guarded by the mutex